cloudify_aws/ecs/cluster.py

```diff
@@ -0,0 +1,237 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+# Stdlib imports
+
+# Third party imports
+
+# Cloudify imports
+from cloudify import ctx
+from cloudify.decorators import operation
+
+# This package imports
+from cloudify_aws import constants
+from cloudify_aws.base import AwsBaseNode, AwsBaseRelationship
+from cloudify_aws.connection import EC2ConnectionClient
+
+
+@operation
+def create(args=None, **_):
+    return Cluster().created(args)
+
+
+@operation
+def delete(args=None, **_):
+    return Cluster().deleted(args)
+
+
+@operation
+def add_container_instance(args=None, **_):
+    return ClusterInstance().associated(args)
+
+
+@operation
+def remove_container_instance(args=None, **_):
+    return ClusterInstance().disassociated(args)
+
+
+class Cluster(AwsBaseNode):
+    def __init__(self, client=None):
+        client = client or EC2ConnectionClient().client3('ecs')
+
+        super(Cluster, self).__init__(
+            constants.ECS_CLUSTER['AWS_RESOURCE_TYPE'],
+            constants.ECS_CLUSTER['REQUIRED_PROPERTIES'],
+            client=client,
+            resource_id_key=constants.ECS_CLUSTER['RESOURCE_ID_KEY'],
+        )
+
+        self.not_found_error = constants.ECS_CLUSTER['NOT_FOUND_ERROR']
+
+    def create(self, args=None, **_):
+        cluster_name = ctx.node.properties['name']
+
+        response = self.execute(
```

Comment:
This is changed in the code, but apparently the new way of marking change requests doesn't update when the code is updated so I'm marking it with a comment so that I can see what I have left to deal with.